cmd/aws-s3-csi-controller/main.go

```diff
@@ -52,6 +56,7 @@ func main() {
 			ImagePullPolicy: corev1.PullPolicy(*mountpointImagePullPolicy),
 		},
 		CSIDriverVersion: version.GetVersion().DriverVersion,
+		IsOpenShift:      isOpenShift(client, log),
```

Comment:
This should be more general, similar to how Mountpoint does [S3 personalities](https://github.com/awslabs/mountpoint-s3/blob/main/mountpoint-s3/src/s3.rs).

Reply:
Thanks for reviewing!

Would something like this make sense? let me know if you have better suggestions regarding naming

```
// ClusterVariant represents different Kubernetes distributions
type ClusterVariant int

const (
	DefaultKubernetes ClusterVariant = iota // Vanilla K8s
	OpenShift                               // OpenShift K8s
)

// MountpointPodUserID returns the appropriate RunAsUser for Mountpoint Pod based on the cluster variant.
func (c ClusterVariant) MountpointPodUserID() *int64 {
	if c == OpenShift {
		// OpenShift clusters automatically assign uid from predefined namespace range
		// https://www.redhat.com/en/blog/a-guide-to-openshift-and-uids
		return nil
	}

	return ptr.To(int64(1000)) // Default non-root Mountpoint user
}
```